src/plugins/query_enhancements/server/routes/resources/routes.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { ILegacyClusterClient, IRouter } from 'opensearch-dashboards/server';
+import {
+  PrometheusManager,
+  PrometheusResourceQuery,
+} from '../../connections/managers/prometheus_manager';
+import { BASE_API } from '../../../common';
+
+export function registerResourceRoutes(router: IRouter, defaultClient: ILegacyClusterClient) {
+  router.get(
+    {
+      path: `${BASE_API}/{dataConnectionType}/{dataConnectionId}/resources/{resourceType}/{resourceName?}`,
+      validate: {
+        params: schema.object({
+          dataConnectionType: schema.oneOf([
+            schema.literal('prometheus'),
+            schema.literal('cloudwatch'),
+          ]),
+          dataConnectionId: schema.string(), // this is the DQS datasource name, which is unique
+          resourceType: schema.oneOf([
+            schema.literal('labels'),
+            schema.literal('label_values'),
+            schema.literal('metric_metadata'),
+            schema.literal('alerts'),
+          ]),
+          resourceName: schema.maybe(schema.string()),
+        }),
+        query: schema.object({}, { unknowns: 'allow' }),
+      },
+    },
+    async (context, request, response) => {
+      const { dataConnectionId, dataConnectionType, resourceType, resourceName } = request.params;
+      if (dataConnectionType === 'prometheus') {
+        const resources = await new PrometheusManager().getResources(context, request, {
```

Comment:
@joshuali925 I know you mentioned this could be added to the route handler context. Can you share where that's defined? Happy to use a shared instance in context in follow up.

Reply:
you can define route context like this https://github.com/opensearch-project/OpenSearch-Dashboards/blob/e8cf0ca4527f55b451e4f9d0af02c82451e0a73e/src/plugins/query_enhancements/server/plugin.ts#L78-L85

then it's accessible in the route handler's `context.<context-name>` variable

It actually must be a shared instance, because the overridden client from other plugins won't work if every time the manager is newly created. might be better if you can make it a singleton

Reply:
Yeah, that's what I was worried about as well. Will export a singleton, register in this context, and then consume from handler context. I'll send out a commit shortly to address.

Reply:
I've updated the prometheus manager so it's only vended as a singleton. I didn't get a chance to update the router context, but will include in next change.